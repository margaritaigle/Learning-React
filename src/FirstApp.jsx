import PropTypes from 'prop-types';
// const newMessage = 'Margarita'; 
// const newMessage2 = 2; 
// const newMessage3 = [1, 2, 3, 'hola', 5];

// const newMessage4 = {
//     name: 'Margarita',
//     age: 23,
// }

// const saludar = (nombre) => {
//     return 'Hola ' + nombre; 
// }

// si debemos imprimir un objeto colocamos JSON.stringify(element)
// SI LA FUNCION ES ASINCRONA, AL OBTENER UN RESULTADO QUE ES UN OBJETO, VA A DAR ERROR
export const FirstApp = ( { title, subtitle, name } ) => {

    //console.log(props);

    return (
        <>
            {/* <h1>{ newMessage }</h1> */}
            {/* <h1>{ saludar('Margarita') }</h1> */}
            {/* <h1>{newMessage2}</h1> Imprime 2 */}
            {/* <h1>{newMessage3}</h1> Imprime '123hola5' */}
            {/* <h4>{JSON.stringify(newMessage4)}</h4> imprime: {"name":"Margarita","age":23} */}
            {/* <h4>{newMessage4.name}</h4> imprime: 'Margarita'*/}
            <h1>{ title }</h1>
            <p>{ subtitle }</p>
            <p>{ name }</p>
        </>
    );
}

// PropTypes deben ser importadas y dado el caso, instaladas. Con ellas puedo asignarle un tipo
// a la propiedad y si debe ser obligatoria (en vite)
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
}

// defaultProps, si no se ingresa nada en la prop, son los valores que se toma por defecto
// seria lo mismo que asignarles un valor en la desestructuracion de la funcion (en los parametros)
FirstApp.defaultProps = {
    name: 'Margarita Iglesias',
    subtitle: 'No hay subtitulo', // al asignarle un valor de otro tipo que no es el asignado, no aparecera error en consola
    title: 'No hay titulo',
}