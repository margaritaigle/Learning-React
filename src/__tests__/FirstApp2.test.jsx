import { render, screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";


describe('Pruebas en FirstApp', () => {

    const title = "Hola, soy Nilda";
    const subTitle = "Soy un subtitulo";

    test('should match with snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('should show the message "Hola, soy Nilda"', () => {

        render(<FirstApp title={title} />);
        //  screen.debug();
        expect(screen.getByText(title)).toBeTruthy();

    });

    test('should show the title in a h1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('should show the subtitle sent by props', () => {
        render(
            <FirstApp
                title={title}
                subtitle={subTitle}
            />
        );
        
        expect( screen.getAllByText(subTitle).length ).toBe(2); 
    });
});