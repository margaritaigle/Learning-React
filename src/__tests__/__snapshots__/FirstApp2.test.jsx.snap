// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Nilda
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Margarita Iglesias
  </p>
</div>
`;
